### PR TITLE
[Chore]: Technical: Translate redusers (vis-state)

### PR DESCRIPTION
**Pull request by HeimEndyd**

### CODE REVIEW

File: src/reducers/vis-state-merger.ts [hunk 1]
```diff
@@ -173,11 +195,14 @@ export function insertLayerAtRightOrder(
...
  */
-export function mergeInteractions(state, interactionToBeMerged) {
+export function mergeInteractions(
+  state: VisState,
+  interactionToBeMerged: NonNullable<ParsedConfig['visState']>['interactionConfig'],
+  fromConfig?: boolean
+): VisState {
   const merged = {};
-  const unmerged = {};
+  const unmerged: {tooltip?: any} = {};
```
Comment (dariaterekhovaae) [inline src/reducers/vis-state-merger.ts:205]: Please find out what type is tooltip and add it there

File: src/reducers/vis-state-merger.ts [hunk 2]
```diff
@@ -276,7 +304,7 @@ export function mergeSplitMaps(state, splitMaps = []) {
  * @param {object} tooltipConfig
  * @return {object} - {mergedTooltip: {}, unmergedTooltip: {}}
  */
-export function mergeInteractionTooltipConfig(state, tooltipConfig = {}) {
+export function mergeInteractionTooltipConfig(state: VisState, tooltipConfig: any = {}) {
```
Comment (dariaterekhovaae) [inline src/reducers/vis-state-merger.ts:307]: The same note: please try to avoid any type

File: src/reducers/vis-state-merger.ts [hunk 3]
```diff
@@ -376,7 +410,12 @@ export function validateSavedLayerColumns(fields, savedCols = {}, emptyCols) {
   return null;
 }
 
-export function validateColumn(column, columns, allFields) {
+export function validateColumn(
+  // TODO: Better types here
+  column: any,
+  columns: any[],
```
Comment (dariaterekhovaae) [inline src/reducers/vis-state-merger.ts:416]: The same note: please avoid any type

File: src/reducers/vis-state-merger.ts [hunk 4]
```diff
@@ -113,13 +133,15 @@ export function mergeLayers(state, layersToMerge, fromConfig) {
     state.layers,
     mergedLayer,
     state.layerOrder,
+    // @ts-expect-error
     preserveLayerOrder
```
Comment (heshan0131) [inline src/reducers/vis-state-merger.ts:137]: `preserveLayerOrder` should be  a array of ids: `string[]`

File: src/reducers/vis-state-updaters.ts [hunk 5]
```diff
@@ -251,7 +513,7 @@ export function updateStateWithLayerAndData(state, {layerData, layer, idx}) {
   };
 }
 
-export function updateStateOnLayerVisibilityChange(state, layer) {
+export function updateStateOnLayerVisibilityChange(state: VisState, layer) {
```
Comment (heshan0131) [inline src/reducers/vis-state-updaters.ts:516]: `export function updateStateOnLayerVisibilityChange(state: VisState, layer: Layer): VisState`

File: src/reducers/vis-state-updaters.ts [hunk 6]
```diff
@@ -1470,7 +1819,7 @@ export function renameDatasetUpdater(state, action) {
  * @param {Object} action action
  * @returns {Object} nextState
  */
-export function closeSpecificMapAtIndex(state, action) {
+export function closeSpecificMapAtIndex(state: VisState, action: any): VisState {
```
Comment (heshan0131) [inline src/reducers/vis-state-updaters.ts:1822]: `action: MapStateActions.ToggleSplitMapUpdaterAction`

**Review by heshan0131 — APPROVED** [review]

File: src/reducers/vis-state-updaters.ts [hunk 7]
```diff
@@ -1574,12 +1926,16 @@ export function loadNextFileUpdater(state) {
   const {loaders, loadOptions} = state;
   return withTask(
     stateWithProgress,
-    // @ts-expect-error
-    makeLoadFileTask(file, nextState.fileLoading.fileCache, loaders, loadOptions)
+    makeLoadFileTask(
+      file,
+      nextState.fileLoading && nextState.fileLoading.fileCache,
```
[the remark this answers is not in the document]
Comment (HeimEndyd) [reply src/reducers/vis-state-updaters.ts:1931]: Type of this field: fileLoading: FileLoading | false.
Property 'fileCache' does not exist on type 'false | FileLoading'.

**Pull request merged**